zerver/lib/actions.py

```diff
@@ -2737,6 +2763,28 @@ def our_person(p):
                         bot.update(event['bot'])
 
         elif event['type'] == 'stream':
+            if event['op'] == 'create':
+                for stream in event['streams']:
+                    stream_name = stream['name']
+                    realm = get_realm(user_profile.realm.domain)
+                    new_stream = get_stream(stream_name, realm)
+                    if not stream['invite_only']:
+                        subscribers = get_subscriber_ids(new_stream)
+                        stream_data = {
+                            'description': new_stream.description,
+                            'stream_id': new_stream.id,
+                            'subscribers':  subscribers,
+                            'invite_only': new_stream.invite_only,
+                            'name': new_stream.name,
+                            'email_address': encode_email_address_helper(new_stream.name, new_stream.email_token)
+                        }
+                        # Add stream to never_subscribed (if not invite_only)
+                        state['never_subscribed'].append(stream_data)
+
+            if event['op'] == 'delete':
+                for stream in event['streams']:
+                    state['stream'] = [s for s in state['stream'] if not s == stream]
```

Comment:
I think `s != stream` is clearer than `not s == stream`. Although other list comprehensions use `not x`, they are calling predicate functions instead of using the built-in equality operator.